pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +106,243 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-// Reconcile reconciles CAPI and MAPI machines for their respective namespaces.
-//
-//nolint:funlen
+// Reconcile reconciles CAPI and MAPI MachineSets for their respective namespaces.
 func (r *MachineSetSyncReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
-	logger := log.FromContext(ctx, "namespace", req.Namespace, "name", req.Name)
+	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	ctx = logr.NewContext(ctx, logger)
```

Comment:
Why this? Is this the new shiny pattern rather than passing `ctx, logger` as the first two arguments?

Reply:
From digging through the code, I think that it's nicer maybe.. WDYT?

It stores the logger with the keys `namespace` and `name` back in the context so any additional `log.FromContext(ctx)` get them for free... 

I thought it may be nicer / cleaner overall as I don't see us storing much in the context otherwise? Maybe it's a use?

Reply:
Works for me, so long as we try to be consistent CC @damdo